### PR TITLE
Create or update credit trade and create comment concurrently

### PR DESCRIPTION
**Pull request by plasticviking**
Changelog:
 - Added 'comment' field to credit_trade POST and PUT and updated serializer to insert comment records
 - Added relevant tests

### CODE REVIEW

**Comment by plasticviking** [comment]
Trello Card 736

File: backend/api/serializers/CreditTrade.py [hunk 1]
```diff
@@ -146,9 +150,37 @@ def validate(self, data):
...
+    def save(self, **kwargs):
+        super().save(**kwargs)
+
+        if 'comment' in self.validated_data \
+                and self.validated_data['comment'] is not None\
+                and len(self.validated_data['comment'].strip()) > 0:
+            comment = CreditTradeComment(
+                credit_trade=self.instance,
+                comment=self.validated_data['comment'],
+                create_user=self.context['request'].user,
```
Comment (amichard) [inline backend/api/serializers/CreditTrade.py]: Just for consistency, can you please do a request = self.context['request'] at the start of the function? Similar to how validate is.

Comment (plasticviking) [reply backend/api/serializers/CreditTrade.py]: Sure

File: backend/api/serializers/CreditTrade.py [hunk 2]
```diff
@@ -328,9 +372,37 @@ def validate(self, data):
...
+    def save(self, **kwargs):
+        super().save(**kwargs)
+
+        if 'comment' in self.validated_data \
+                and self.validated_data['comment'] is not None \
+                and len(self.validated_data['comment'].strip()) > 0:
+            comment = CreditTradeComment(
+                credit_trade=self.instance,
+                comment=self.validated_data['comment'],
+                create_user=self.context['request'].user,
```
Comment (amichard) [inline backend/api/serializers/CreditTrade.py]: Same comment as the one above.

Comment (plasticviking) [reply backend/api/serializers/CreditTrade.py]: Addressed in latest commit

**Pull request merged**